prov/util/src/util_ep.c

```diff
@@ -129,6 +129,21 @@ int ofi_endpoint_init(struct fid_domain *domain, const struct util_prov *util_pr
 int ofi_endpoint_close(struct util_ep *util_ep)
 {
 	fastlock_destroy(&util_ep->lock);
+
+	if (util_ep->tx_cq) {
+		ofi_atomic_dec32(&util_ep->tx_cq->ref);
```

Comment:
This should come after fid_list_remove. Only after that we are not using the cq anymore.